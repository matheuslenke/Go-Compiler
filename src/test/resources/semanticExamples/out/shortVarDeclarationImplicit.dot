digraph {
graph [ordering="out"];
node0[label="source_file"];
node1[label="import_spec"];
node2[label="Println@1"];
node3[label="parameters"];
node4[label="(string) arg@2"];
node3 -> node4;
node2 -> node3;
node5[label="result"];
node2 -> node5;
node1 -> node2;
node6[label="Scanln@3"];
node7[label="parameters"];
node8[label="(string) arg@4"];
node7 -> node8;
node6 -> node7;
node9[label="result"];
node6 -> node9;
node1 -> node6;
node0 -> node1;
node10[label="(func) testeAtribuicaoImplicitaCurta@5"];
node11[label="parameters"];
node10 -> node11;
node12[label="block"];
node13[label="var_list"];
node14[label=":="];
node15[label="(int) a@6"];
node14 -> node15;
node16[label="(int) int@1"];
node14 -> node16;
node13 -> node14;
node12 -> node13;
node17[label="var_list"];
node18[label=":="];
node19[label="(float) b@7"];
node18 -> node19;
node20[label="(float) real@1.5"];
node18 -> node20;
node17 -> node18;
node12 -> node17;
node21[label="var_list"];
node22[label=":="];
node23[label="(string) c@8"];
node22 -> node23;
node24[label="(string) string@0"];
node22 -> node24;
node21 -> node22;
node12 -> node21;
node25[label="var_list"];
node26[label=":="];
node27[label="(bool) d@9"];
node26 -> node27;
node28[label="(bool) bool@true"];
node26 -> node28;
node25 -> node26;
node12 -> node25;
node29[label="var_list"];
node30[label=":="];
node31[label="(bool) e@10"];
node30 -> node31;
node32[label="(bool) bool@false"];
node30 -> node32;
node29 -> node30;
node12 -> node29;
node33[label="var_list"];
node34[label="="];
node35[label="(int) f@11"];
node34 -> node35;
node36[label="(int) int@1"];
node34 -> node36;
node33 -> node34;
node37[label="="];
node38[label="(int) g@12"];
node37 -> node38;
node39[label="(int) int@2"];
node37 -> node39;
node33 -> node37;
node12 -> node33;
node40[label="var_list"];
node41[label="="];
node42[label="(float) h@13"];
node41 -> node42;
node43[label="(float) real@1.5"];
node41 -> node43;
node40 -> node41;
node44[label="="];
node45[label="(bool) i@14"];
node44 -> node45;
node46[label="(bool) bool@true"];
node44 -> node46;
node40 -> node44;
node12 -> node40;
node47[label="var_list"];
node48[label="="];
node49[label="(bool) j@15"];
node48 -> node49;
node50[label="(bool) bool@false"];
node48 -> node50;
node47 -> node48;
node51[label="="];
node52[label="(string) k@16"];
node51 -> node52;
node53[label="(string) string@1"];
node51 -> node53;
node47 -> node51;
node12 -> node47;
node10 -> node12;
node0 -> node10;
}
