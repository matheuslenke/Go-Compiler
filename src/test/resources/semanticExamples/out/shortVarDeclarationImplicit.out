[INFO] Scanning for projects...
[INFO] 
[INFO] ----------------< br.ufes.edu.compiladores:go-compiler >----------------
[INFO] Building Compilador de GO 1.0-SNAPSHOT
[INFO] --------------------------------[ jar ]---------------------------------
[INFO] 
[INFO] --- exec-maven-plugin:3.0.0:java (default-cli) @ go-compiler ---
PARSE SUCCESSFUL



Strings table:
Entry 0: Hello World
Entry 1: Bye World




Variables table:
Entry 0 -- name: fmt, line: 3, type: no_type, scopeName: global
Entry 1 -- name: Println, line: 3, type: no_type, scopeName: global
Entry 2 -- name: arg, line: 3, type: string, scopeName: Println
Entry 3 -- name: Scanln, line: 3, type: no_type, scopeName: global
Entry 4 -- name: arg, line: 3, type: string, scopeName: Scanln
Entry 5 -- name: testeAtribuicaoImplicitaCurta, line: 5, type: func, scopeName: global
Entry 6 -- name: a, line: 6, type: int, scopeName: testeAtribuicaoImplicitaCurta
Entry 7 -- name: b, line: 7, type: float, scopeName: testeAtribuicaoImplicitaCurta
Entry 8 -- name: c, line: 8, type: string, scopeName: testeAtribuicaoImplicitaCurta
Entry 9 -- name: d, line: 9, type: bool, scopeName: testeAtribuicaoImplicitaCurta
Entry 10 -- name: e, line: 10, type: bool, scopeName: testeAtribuicaoImplicitaCurta
Entry 11 -- name: f, line: 12, type: int, scopeName: testeAtribuicaoImplicitaCurta
Entry 12 -- name: g, line: 12, type: int, scopeName: testeAtribuicaoImplicitaCurta
Entry 13 -- name: h, line: 13, type: float, scopeName: testeAtribuicaoImplicitaCurta
Entry 14 -- name: i, line: 13, type: bool, scopeName: testeAtribuicaoImplicitaCurta
Entry 15 -- name: j, line: 14, type: bool, scopeName: testeAtribuicaoImplicitaCurta
Entry 16 -- name: k, line: 14, type: string, scopeName: testeAtribuicaoImplicitaCurta




[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time:  0.710 s
[INFO] Finished at: 2022-02-24T20:53:58-03:00
[INFO] ------------------------------------------------------------------------
